services/core/shared/core/delegates.js

```diff
@@ -249,6 +249,34 @@ const reload = async () => {
 	await computeDelegateStatus();
 };
 
+Signals.get('newBlock').add(async data => {
+	const updatedDelegateAddresses = [];
+	const [block] = data.data;
+	const dposModuleId = 5;
+	const voteDelegateAssetId = 1;
+	// to support old versions
+	if (block && block.payload) {
+		block.payload.forEach(tx => {
+			if (tx.moduleID === dposModuleId && tx.assetID === voteDelegateAssetId) {
+				tx.asset.votes.forEach(vote => updatedDelegateAddresses
+					.push(coreApi.getBase32AddressFromHex(vote.delegateAddress)));
+			}
+		});
+		const { data: updatedDelegateAccounts } = await coreApi
+			.getAccounts({ addresses: updatedDelegateAddresses });
+		updatedDelegateAccounts.forEach(delegate => {
+			const indexOfDelegate = delegateList.findIndex(acc => acc.address === delegate.address);
+			if (indexOfDelegate === -1) {
+				delegateList.push(delegate);
+			} else {
+				delegateList[indexOfDelegate] = delegate;
+			}
+		});
+		await computeDelegateRank();
+		await computeDelegateStatus();
+	}
+});
```

Comment:
Delegate `rank` would vary only when the `voteWeight` changes and the `status` gets updated at the beginning of a new round. 

```suggestion
// Keep the delegate cache up-to-date
Signals.get('newBlock').add(async data => {
	const dposModuleId = 5;
	const voteDelegateAssetId = 1;

	const updatedDelegateAddresses = [];

	const [block] = data.data;
	if (block && block.payload) {
		block.payload.forEach(tx => {
			if (tx.moduleID === dposModuleId && tx.assetID === voteDelegateAssetId) {
				tx.asset.votes.forEach(vote => updatedDelegateAddresses
					.push(coreApi.getBase32AddressFromHex(vote.delegateAddress)));
			}
		});

		const { data: updatedDelegateAccounts } = await coreApi
			.getAccounts({ addresses: updatedDelegateAddresses });

		updatedDelegateAccounts.forEach(delegate => {
			const delegateIndex = delegateList.findIndex(acc => acc.address === delegate.address);
			if (delegateIndex === -1) delegateList.push(delegate);
			else delegateList[delegateIndex] = delegate;
		});

		// Rank is impacted only when a delegate gets (un-)voted
		if (updatedDelegateAddresses.length) await computeDelegateRank();
	}
});
```